### ISSUE TITLE
Keys defined as strings return nil values

### ISSUE THREAD

**Issue by jromigh** [aetherknight/recursive-open-struct#28]
In v0.6.2, if a hash where keys are set as strings (as opposed to symbols) is passed into recursive open struct, all values retrieved are returned as nil.

Here is my code:
```ruby
require 'recursive-open-struct'

# String type
ros = RecursiveOpenStruct.new( { "foo" => { "bar" => "val" }, "another_key" => "another_value", "hello" => "world" } )
puts ros.inspect
puts "foo = #{ros.foo}"
puts "another_key = #{ros.another_key}"
puts "hello = #{ros.hello}"

# Symbol type
ros = RecursiveOpenStruct.new( { :foo => { :bar => "val" }, :another_key => "another_value", :hello => "world" } )
puts ros.inspect
puts "foo = #{ros.foo}"
puts "another_key = #{ros.another_key}"
puts "hello = #{ros.hello}"
```

In v0.6.2, this is what I get in console:
```
C:\testing_recursive_ostruct>bundle exec ruby testing_recursive_ostruct.rb
#<RecursiveOpenStruct foo={"bar"=>"val"}, another_key="another_value", hello="world">
foo =
another_key =
hello =
#<RecursiveOpenStruct foo={:bar=>"val"}, another_key="another_value", hello="world">
foo = #<RecursiveOpenStruct bar="val">
another_key = another_value
hello = world
```

In v0.6.1, this is what I get in console:
```
C:\testing_recursive_ostruct>bundle exec ruby testing_recursive_ostruct.rb
#<RecursiveOpenStruct foo={"bar"=>"val"}, another_key="another_value", hello="world">
foo = #<RecursiveOpenStruct bar="val">
another_key = another_value
hello = world
#<RecursiveOpenStruct foo={:bar=>"val"}, another_key="another_value", hello="world">
foo = #<RecursiveOpenStruct bar="val">
another_key = another_value
hello = world
```

**Comment by celsworth** [comment]
Can confirm this, also just got bitten by it.

0.6.2:
```ruby
RecursiveOpenStruct.new('foo' => 1).foo
=> nil
```

0.6.1:
```ruby
RecursiveOpenStruct.new('foo' => 1).foo
=> 1
```

**Comment by thiagogsr** [comment]
https://github.com/aetherknight/recursive-open-struct/pull/30

**Comment by gerrywastaken** [comment]
Thanks for the fix @thiagogsr.

**Comment by thiagogsr** [comment]
You are welcome

**Comment by aetherknight** [comment]
Merged #30 into master, I will cut a patch release soon.

**Comment by aetherknight** [comment]
Version 0.6.3 should now be on rubygems, and it should contain the fix in #30.
